contracts/DependencyRegistryV1.sol

```diff
@@ -0,0 +1,1074 @@
+
+// Created By: Art Blocks Inc.
+
+import "./interfaces/0.8.x/IAdminACLV0.sol";
+import "./interfaces/0.8.x/IDependencyRegistryCompatibleV0.sol";
+import "./interfaces/0.8.x/IDependencyRegistryV0.sol";
+
+import "@openzeppelin-4.7/contracts/utils/Strings.sol";
+import "@openzeppelin-4.7/contracts/utils/structs/EnumerableSet.sol";
+import "@openzeppelin-4.8/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "@openzeppelin-4.8/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin-4.5/contracts/utils/math/SafeCast.sol";
+
+import {IContractScript} from "scripty.sol/contracts/scripty/IContractScript.sol";
+import {AddressChunks} from "./AddressChunks.sol";
+import {IScriptyBuilder, WrappedScriptRequest} from "scripty.sol/contracts/scripty/IScriptyBuilder.sol";
+
+import "./libs/0.8.x/BytecodeStorage.sol";
+import "./libs/0.8.x/Bytes32Strings.sol";
+
+import "hardhat/console.sol";
+
+/**
+ * @title Art Blocks Dependency Registry, V0.
+ * @author Art Blocks Inc.
+ * @notice Privileged Roles and Ownership:
+ * Permissions managed by ACL contract. If/when we ever call
+ * renounceOwnership() this will becom a frozen, immutable registry
+ * as no upgrades will be possible.
+ * This contract This contract is intended to be an auxiliary reference registry
+ * to our non-upgradeable and immutable ERC-721 conforming core contracts,
+ * and has been made upgradeable as we expect its required functionality in
+ * relation to the Art Blocks ecosystem to evolve over time.
+ */
+contract DependencyRegistryV1 is
+    Initializable,
+    OwnableUpgradeable,
+    IDependencyRegistryV0,
+    IContractScript
+{
+    using BytecodeStorage for string;
+    using BytecodeStorage for address;
+    using Bytes32Strings for bytes32;
+    using Strings for uint256;
+    using EnumerableSet for EnumerableSet.Bytes32Set;
+    using EnumerableSet for EnumerableSet.AddressSet;
+    using SafeCast for uint24;
+
+    uint8 constant AT_CHARACTER_CODE = uint8(bytes1("@")); // 0x40
+
+    /// admin ACL contract
+    IAdminACLV0 public adminACLContract;
+
+    struct Dependency {
+        string preferredCDN;
+        mapping(uint256 => string) additionalCDNs;
+        string preferredRepository;
+        mapping(uint256 => string) additionalRepositories;
+        string referenceWebsite;
+        // mapping from script index to address storing script in bytecode
+        mapping(uint256 => address) scriptBytecodeAddresses;
+        uint24 additionalCDNCount;
+        uint24 additionalRepositoryCount;
+        uint24 scriptCount;
+    }
+
+    EnumerableSet.Bytes32Set private _dependencyTypes;
+    mapping(bytes32 => Dependency) dependencyDetails;
+
+    EnumerableSet.AddressSet private _supportedCoreContracts;
+    mapping(address => mapping(uint256 => bytes32)) projectDependencyTypeOverrides;
+
+    IScriptyBuilder public scriptyBuilder;
+    address public ethFsAddress;
+
+    modifier onlyNonZeroAddress(address _address) {
```

Comment:
nit - do you mind switching these to internal functions? recently switched over all active contracts to not use modifiers as a best practice, described here: https://github.com/ArtBlocks/artblocks-contracts/blob/main/GUIDELINES.md